go-controller/pkg/cni/ovs.go

```diff
@@ -110,66 +111,86 @@ func ofctlExec(args ...string) (string, error) {
 	return stdoutStr, nil
 }
 
-func waitForPodFlows(mac string, ifAddrs []*net.IPNet, ifaceName, ifaceID string) error {
+func isIfaceIDSet(ifaceName, ifaceID string) (bool, error) {
+	// ensure the OVS interface is still active. It may have been cleared by a subsequent CNI ADD
+	// and if so, there's no need to keep checking for flows
+	out, err := ovsGet("Interface", ifaceName, "external-ids", "iface-id")
+	if err != nil {
+		return false, nil
+	}
+	if out != ifaceID {
+		return false, fmt.Errorf("OVS sandbox port %s is no longer active (probably due to a subsequent "+
+			"CNI ADD)", ifaceName)
+	}
+
+	return true, nil
+}
+
+func doPodFlowsExist(mac string, ifAddrs []*net.IPNet) (bool, error) {
```

Comment:
Food for thought, is this a hard requirement or can we relax it and hand over the responsibility to OVN/OVN controller to reconcile on this and make that the flows are set up correctly... flows must be there sooner or later, or can it happen that it never make it?

Reply:
we cannot relax the check, and did you read the bz below? :)

Reply:
Dumitru is working on a better method right now actually; but that won't be available in acutal builds for a while. We've been using this workaround for a while.

Reply:
sorry, let me rephrase, what I'm trying to mean is that I see this as a 2 part process, 
1. binding the port in the node, so the local plumbing is ready
2. certify thath OVN has everything set, OVN install the flows and everything it need to provide connectivity to thepod
The bz talks about a way to check 2 from the Node, what I'm asking is, if we can skip 2. once 1. is ready, because it may possible to ask the OVN-Kube controller to reconcile 2 , checking in a loop until OVN finish (naive question, don't know the internals of the OVN flows setup)